modules/nf-core/md5sum/main.nf

```diff
@@ -22,10 +22,16 @@ process MD5SUM {
     def prefix = task.ext.prefix ?: "${meta.id}"
 
     """
+    IFS=\$(echo -en "\n\b")
+    for FILE in $files
+    do
+
     md5sum \\
         $args \\
-        ${file} \\
-        > ${file}.md5
+        \${FILE} \\
+        > "\${FILE}.md5"
+
+    done
```

Comment:
Here's my suggestion based on Matthieu's suggestion
```bash
    input:
    tuple val(meta), path(files)
    val as_separate_files

    script:
    if ( as_separate_files ) {
        """
        find -L * -type f \\
            ! -name '*.md5' \\
            -exec sh -c 'md5sum "\$1" > "\$1.md5"' _ "{}" \\;
        """
    } else {
        """
        find -L * -type f \\
            ! -name '*.md5' \\
            -exec md5sum "{}" + \\
            > all.md5
        """
    }

    output:
    tuple val(meta), path("*.md5"), emit: checksum

```
Only caveat is that `files` must be files and not directories, otherwise the first instance will write out md5's to input subdirectories.
This should also handle spaces.